sycl/test/check_device_code/group_load.cpp

```diff
@@ -41,6 +40,119 @@ using plain_global_ptr = typename sycl::detail::DecoratedType<
 template SYCL_EXTERNAL void sycl::ext::oneapi::experimental::group_load<
     sycl::sub_group, plain_global_ptr<int>, int, naive_blocked>(
     sycl::sub_group, plain_global_ptr<int>, int &, naive_blocked);
+
+// Check that optimized implementation is selected.
+template SYCL_EXTERNAL void sycl::ext::oneapi::experimental::group_load<
+    sycl::sub_group, plain_global_ptr<int>, int, opt_blocked>(
+    sycl::sub_group, plain_global_ptr<int>, int &, opt_blocked);
```

Comment:
This needs some manual post-processing to align CHECKs with the corresponding instantiation.

Reply:
I would rather not, manual editing introduce maintenance effort .

Reply:
I can't agree with this. Yes, there needs to be an improvement, but simply doing what this PR is doing is wrong.

Reply:
Can you be more specific what is `wrong`? Yes, it affects some readability, but I can't think of where is wrong.

Reply:
I think readability equals maintainability and drastic decrease in that is wrong.

Reply:
Fine. I will update it for readability this time. So far, the change to this file for maintenance and feature work is about 50-50. If we get more maintenance changes later, we should consider improve it.

```
* 07b772877808 - (2024-12-09) Regen group_load/store after 7954a0514ba7 - Jinsong Ji (HEAD -> regengroupmem, jsji/regengroupmem)
* 9908dcd9622a - (2024-11-13) Merge from 'sycl' to 'sycl-web' (5 commits) - iclsrc
* 61d963b7b04f - (2024-11-13) [SYCL] Use inheritance to implement oneapi properties (#16024) - aelovikov-intel
* 797177b9abc6 - (2024-11-13) Merge remote-tracking branch 'upstream/llvmspirv_pulldown' into HEAD - Sarnie, Nick
* 28165194bda1 - (2024-11-08) [SYCL][Test] Update attributes in tests after  69a798a 095d49d - Jinsong Ji
* 1d995c64c4ae - (2024-09-09) Revert "Partially revert "Add nuw in GEP for check_device_code tests"" - Evgenii Kudriashov
* c930dbbcb781 - (2024-08-30) Partially revert "Add nuw in GEP for check_device_code tests" - Evgenii Kudriashov
* 34997a28750b - (2024-08-28) Add nuw in GEP for check_device_code tests - Evgenii Kudriashov
* 1f8315ccdd0e - (2024-05-10) Merge from 'sycl' to 'sycl-web' - iclsrc
* 48a0ff5b4b5b - (2024-05-09) [SYCL] Optimize SG group_load via BlockReadINTEL in simple cases (#13673) - aelovikov-intel
* bc18ad16e014 - (2024-05-08) [SYCL] Use `-fno-discard-value-names` in relevant `check_device_code` tests (#13716) - aelovikov-intel
* 967134c565b0 - (2024-05-08) [SYCL][E2E] Limit check_device_code/group_load.cpp to asserts build (#13711) - aelovikov-intel
* f56d7d7ffd79 - (2024-05-07) [SYCL] Add a test for generated device code for the group_load_store extension (#13671) - aelovikov-intel 
```

Reply:
While I agree in general, the ratio isn't accurate because there are "silent" benefits when this test catches regression during refactoring.

Reply:
I agree that these tests are catching regressions during development indeed. So yeah, it is hard to measure the ratio. But in any case, it would be great if we can find some way to lower the maintenance effort while keeping the readability. Thank you!